promql/functions.go

```diff
@@ -1385,19 +1386,27 @@ func init() {
 		delete(parser.Functions, "xincrease")
 		delete(parser.Functions, "xrate")
 		fmt.Println("Successfully replaced rate & friends with xrate & friends (and removed xrate & friends function keys).")
-	case "2":
+
+	case "2", "y", "Y":
+		origDeltaFunction := *parser.Functions["delta"]
+		origDeltaFunction.Name = "orig_delta"
+		parser.Functions["orig_delta"] = &origDeltaFunction
+		FunctionCalls["orig_delta"] = FunctionCalls["delta"]
```

Comment:
Should we add this to `x` case as well?

Reply:
Possibly. I didn't really want to change the semantics that had for `=1`, which I thought were ugly:
1. They replace the original functions with the x* versions.
2. You cannot call the orig ones, nor can you call the x ones.

I suppose I could leave the above for `=1`, and make a separate `=x` version that did what I found cleaner? Basically the same thing as `=y` but applied to `x`?

In both cases of `=x` and `=y`, you'd be able to call any explicit version that you wanted:
- `orig_`
- `x*`
- `y*`

The only difference would be what you got when you called a function by its OG name.

This doesn't sound too bad, and not too hard either. Just a copy-paste I think...

Reply:
Done: https://github.com/Invoca/prometheus/pull/5/commits/3cfcd15622bab2945e4c254a809407f49cf6f181